crates/cheatcodes/src/evm/fork.rs

```diff
@@ -243,8 +244,15 @@ impl Cheatcode for rpcCall {
             foundry_common::block_on(provider.raw_request(method.clone().into(), params_json))
                 .map_err(|err| fmt_err!("{method:?}: {err}"))?;
 
-        let result_as_tokens = crate::json::json_value_to_token(&result)
-            .map_err(|err| fmt_err!("failed to parse result: {err}"))?;
+        let result_as_tokens = match crate::json::json_value_to_token(&result)
+            .map_err(|err| fmt_err!("failed to parse result: {err}"))?
+        {
+            DynSolValue::FixedBytes(bytes, _) => {
+                // converted fixed bytes to bytes to prevent evm encoding issues: <https://github.com/foundry-rs/foundry/issues/8287>
+                DynSolValue::Bytes(bytes.to_vec())
```

Comment:
this will be vec![_; 32] for bytes31, bytes30 ... is this right?

Reply:
`bytes.as_slice()[..size].to_vec()`?

Reply:
yeah